homeassistant/components/media_player/intent.py

```diff
@@ -0,0 +1,48 @@
+"""Intents for the media_player integration."""
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    SERVICE_MEDIA_NEXT_TRACK,
+    SERVICE_MEDIA_PAUSE,
+    SERVICE_MEDIA_PLAY,
+    SERVICE_VOLUME_SET,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import intent
+import homeassistant.helpers.config_validation as cv
+
+from . import ATTR_MEDIA_VOLUME_LEVEL, DOMAIN
+
+INTENT_MEDIA_PAUSE = "HassMediaPause"
+INTENT_MEDIA_UNPAUSE = "HassMediaUnpause"
+INTENT_MEDIA_NEXT = "HassMediaNext"
+INTENT_SET_VOLUME = "HassSetVolume"
+
+
+async def async_setup_intents(hass: HomeAssistant) -> None:
+    """Set up the media_player intents."""
+    intent.async_register(
+        hass,
+        intent.ServiceIntentHandler(INTENT_MEDIA_UNPAUSE, DOMAIN, SERVICE_MEDIA_PLAY),
+    )
+    intent.async_register(
+        hass,
+        intent.ServiceIntentHandler(INTENT_MEDIA_PAUSE, DOMAIN, SERVICE_MEDIA_PAUSE),
+    )
+    intent.async_register(
+        hass,
+        intent.ServiceIntentHandler(
+            INTENT_MEDIA_NEXT, DOMAIN, SERVICE_MEDIA_NEXT_TRACK
+        ),
+    )
+    intent.async_register(
+        hass,
+        intent.ServiceIntentHandler(
+            INTENT_SET_VOLUME,
+            DOMAIN,
+            SERVICE_VOLUME_SET,
+            extra_slot_names={ATTR_MEDIA_VOLUME_LEVEL},
+            extra_slot_schema={vol.Required(ATTR_MEDIA_VOLUME_LEVEL): cv.small_float},
```

Comment:
Why don't we just accept 0..100 and divide it by 100 before passing it to the service? I don't think that we should let our service design decisions leak into the intent?

```python
vol.All(vol.Range(min=0, max=100), lambda val: val/100)
```

Reply:
That will also make it easier if ever AI needs to generate intent calls.

Reply:
I'm happy to do that  :+1: